### PR TITLE
pd_client: report term in region heartbeat

### PR DESCRIPTION
**Pull request by shafreeck**
Signed-off-by: Shafreeck Sea <shafreeck@gmail.com>

## What have you changed? (mandatory)

Report the raft term in region heartbeat.

PD needs this information to tell if the leader of a region has changed. If it is, the stats TiKV reported should be carefully handled because the stats may differ in huge.

## What are the type of changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Improvement (a change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

- unit tests

## Does this PR affect documentation (docs) or release note? (mandatory)

NO

## Does this PR affect `tidb-ansible` update? (mandatory)

NO

## Refer to a related PR or issue link (optional)

- https://github.com/pingcap/kvproto/pull/410

### CODE REVIEW

**Comment by shafreeck** [comment]
> kvproto is already updated before?

@Connor1996 Yes, it is.

File: components/test_raftstore/src/pd.rs [hunk 1]
```diff
@@ -556,6 +556,7 @@ impl Cluster {
 
     fn region_heartbeat(
         &mut self,
+        _term: u64,
```
Comment (Connor1996) [inline components/test_raftstore/src/pd.rs]: maybe should add some logics about `term`

Comment (shafreeck) [reply components/test_raftstore/src/pd.rs]: I tried, but I have no idea what to add. Would you like to give some suggestions?

Comment (nolouch) [reply components/test_raftstore/src/pd.rs]: Maybe like `fn handle_heartbeat_version`.

**Comment by Connor1996** [comment]
should it cherry-pick to release-3.0?

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -2298,6 +2303,7 @@ impl Peer {
 
     pub fn heartbeat_pd<T, C>(&mut self, ctx: &PollContext<T, C>) {
         let task = PdTask::Heartbeat {
+            term: self.get_last_term(),
```
Comment (overvenus) [inline src/raftstore/store/peer.rs]: Why not `self.term()`?

**Pull request merged**